exercises/arrays/allPositive/allPositive.js

```diff
@@ -13,13 +13,20 @@
  */
 function allPositive(array) {
   // This is your job. :)
+  for (let element in array){
```

Comment:
Ah, good catch. In your comment yesterday you mentioned for ... of instead of using for(...;...;...). So, I accidentally started using for ... in during every loop thinking I was doing for ... of. Thank you!

Reply:
Yeah, most languages that let you iterate over elements have `for thing in collection`. JS had `for ... in ...` from the beginning but it did something else.

Technically what it does it iterate over the properties of an object. If you have an object that has properties added to it from elsewhere — not uncommon if you're using a library — those will appear when you use `for ... in`.

Try this (which might seem weird but is valid JS):

```js
let array = [10, 20, 30, 40];
array._special = true;

for (let i in array) {
  console.log('i is:', i);
  console.log('array[i] is:', array[i]);
}

for (let i = 0; i < array.length; i++) {
  console.log('i is:', i);
  console.log('array[i] is:', array[i]);
}

for (let item of array) {
  console.log('item is:', item);
}
```